core/src/main/java/org/springframework/security/converter/RsaKeyConverters.java

```diff
@@ -72,8 +72,8 @@ private RsaKeyConverters() {
 		return (source) -> {
 			List<String> lines = readAllLines(source);
 			Assert.isTrue(!lines.isEmpty() && lines.get(0).startsWith(PKCS8_PEM_HEADER),
-					"Key is not in PEM-encoded PKCS#8 format, please check that the header begins with -----"
-							+ PKCS8_PEM_HEADER + "-----");
+					"Key is not in PEM-encoded PKCS#8 format, please check that the header begins with "
```

Comment:
It would be nice to improve the two tests for these two methods to verify the message as well. One example would be:
```java
@Test
public void pkcs8WhenConvertingPkcs1PrivateKeyThenIllegalArgumentException() {
	assertThatIllegalArgumentException().isThrownBy(() -> this.pkcs8.convert(toInputStream(PKCS1_PRIVATE_KEY)))
				.withMessage("Key is not in PEM-encoded PKCS#8 format, please check that the header begins with " +
						"-----BEGIN PRIVATE KEY-----");
}
```
The same goes for `x509` method.